bin/peekaboo-util.py

```diff
@@ -66,6 +66,47 @@ def send_receive(self, request):
         logger.debug('Received from peekaboo: %s', buf)
         return buf
 
+    def ping(self):
+        """ Send ping request to daemon and optionally print response. """
+        ping = json.dumps([{"request": "ping"}])
+
+        logger.debug("Sending ping...")
+        try:
+            response = self.send_receive(ping)
+        except socket.error as error:
+            logger.error("Error communicating with daemon: %s", error)
+            return 2
+
+        pong = None
+        for line in response.splitlines():
+            try:
+                # try to parse and stop at first thing that parses
+                pong = json.loads(line)
+                break
+            except ValueError as error:
+                # FIXME: daemon talks a mix of plain text and JSON. So for now
```

Comment:
@jack28: Here's another reason why we maybe want a separate command socket with a clean JSON-only protocol. Altenratively we could expedite converting the current protocol on the client socket to JSON. That would mean some tinkering with the amavis plugin but is certainly doable and on our to-do list anyway. What do you think?

Reply:
Like you said, we want it anyway.

Sooner rather than later we will move to a network socket and won't be able to rely on file permissions to control who accesses the command socket.

Reply:
Okay, expedited conversion of client protocol to JSON is the way we go.